pydantic/types.py

```diff
@@ -1238,7 +1239,7 @@ class Model(BaseModel):
 
 @_dataclasses.dataclass
 class PathType:
-    path_type: Literal['file', 'dir', 'new']
+    path_type: Literal['file', 'dir', 'new', 'socket']
```

Comment:
```suggestion
    if sys.platform != "win32":
        path_type: Literal['file', 'dir', 'new', 'socket']
    else:
        path_type: Literal['file', 'dir', 'new']
```

Reply:
i am suprised that conditional attributes on dataclasses are a thing

Reply:
But anyway, turns out you can also have sockets on Windows, so let's not add any conditions.

Reply:
Interesting, ok. Closing these conversations then.